frr-reloader/frr-reloader.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+cleanup() {
+  echo "Caught an exit signal.."
+  rm -f "$PIDFILE"
+  rm -f "$PIPE_FILE"
+  kill_sleep
+  exit
+}
+
+reload_and_wait() {
+  reload_frr & 
+  cat "$PIPE_FILE" > /dev/null
+}
+
+reload_frr() {
+  echo "Caught SIGHUP! Reloading FRR.."
+  kill_sleep
+
+  echo "Checking the configuration file syntax"
+  if ! python3 /usr/lib/frr/frr-reload.py --test --stdout "$FILE_TO_RELOAD" ; then
+    echo "Syntax error spotted: aborting.."
+    echo "" > "$PIPE_FILE"
+    return
+  fi
+
+  echo "Applying the configuration file"
+  if ! python3 /usr/lib/frr/frr-reload.py --reload --overwrite --stdout "$FILE_TO_RELOAD" ; then
+    echo "Failed to fully apply configuration file: aborting.."
```

Comment:
It does not actually abort - changed the message to clarify it. The script exiting with an error here could indicate that some section in the config couldn't be applied against frr. No roll-back or abort happens, the script applies what it can from the config file.